### ISSUE TITLE
t.p.dir.isDirectory() segfaults with wrong args

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7465]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)| @warner reported|
|-|-|
|Trac ID|trac#121|
|Type|defect|
|Created|2003-07-25 00:49:40Z|

<details><summary>Searchable metadata</summary>

```
trac-id__121 121
type__defect defect
reporter__warner warner
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1059094180000000 1059094180000000
changetime__1465028419681803 1465028419681803
version__ 
owner__ 
cc__exarkun cc__warner
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
twisted.python.dir.isDirectory("/home/warner") segfaults.

It appears that isDirectory and friends are meant to be
instance methods of the object (a tuple) that comes out of
the iterator. However, dir(dir) shows they are top-level
functions, so the casual user might assume they could be
called just like os.path.isdir(). The C code does no
checking of the argument to make sure it is a tuple, and
blows up when it gets something that isn't.

Not sure what to suggest here.. the cleanest answer is to
have the iterator create a new object (which behaves like a
tuple) so that isDirectory could be a proper instance method
(removing the concern that it could be called incorrectly).
However that probably has a performance penalty, unfortunate
since the whole point of dir.c is to speed up directory
traversal.

The quicker hack might be to just verify that args[0] is a
tuple with an int as the second member.

In any case isDirectory and friends should be documented
somewhere so people know to not call them like os.path.isdir

 -Brian

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
is*() re-implemented in Python, segfaults begone!
documentation added too, though somewhat minimal.

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

In changeset 1bcffbec9d4ac0011e575efef6e07f2494a29622
```
#!CommitTicketReference repository="" revision="1bcffbec9d4ac0011e575efef6e07f2494a29622"
Merge pull request #121 from rodrigc/8405-rodrigc-print: Convert from print statement to print() function in twisted/conch

Author: rodrigc
Reviewer: cyli
Fixes: #8405

Convert from print statement to print() function in twisted/conch
```
